### ISSUE TITLE
Add Naive Bayes

### ISSUE THREAD

**Issue by param087** [param087/swiftML#12]
### **1. Bernoulli Naive Bayes Classifier -**
  - _BernoullilNB( alpha:1.0)_ -
      - **alpha**(Float) - Smoothing parameter.

  ### **API Methods -**
   - **fit(data: , labels: )** - Fit naive bayes classifier.
   - **prediction(for: )** - Perform classification on input data.
   - **predictLogProba(data: )** - Returns log probability of input data. 
   - **score(data: , labels: )** - Returns the mean accuracy.

  ### **References** 
   - https://en.wikipedia.org/wiki/Naive_Bayes_classifier
   - https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.BernoulliNB.html
   - https://nlp.stanford.edu/IR-book/html/htmledition/the-bernoulli-model-1.html

### **2. Multinomial Naive Bayes Classifier -**
 - _MultinomialNB( alpha:1.0)_ -
   - **alpha**(Float) - Smoothing parameter.

  ### **API Methods -**
   - **fit(data: , labels: )** - Fit naive bayes classifier
   - **prediction(for: )** - Perform classification on data.
   - **predictLogProba(data: )** - Returns log probability of data.  
   - **score(data: , labels: )** - Returns the mean accuracy.

  ### **References** 
   - https://en.wikipedia.org/wiki/Naive_Bayes_classifier
   - https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.MultinomialNB.htmll
   - https://nlp.stanford.edu/IR-book/html/htmledition/naive-bayes-text-classification-1.html

 ### **3. Gaussian Naive Bayes Classifier -**
  -  _GaussianNB( )_ -
   

 ###  **API Methods -**
   - **fit(data: , labels: )** - Fit naive bayes classifier
   - **prediction(for: )** - Perform classification on data.
   - **prob(data: )** - Return gaussian probability in log.
   - **predictLogProba(data: )** - Returns log probability of data. 
   - **score(data:  ,labels: )** - Returns the mean accuracy.

 ###  **References** 
  - https://en.wikipedia.org/wiki/Naive_Bayes_classifier
  - https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.GaussianNB.html
